docs/src/content/docs/datamessages.mdx

```diff
@@ -0,0 +1,90 @@
+---
+title: "Data Messages"
+description: Protocol documentation for messages exchanged between client and server during Ultravox calls.
+---
+
+Data messages are used to communicate state, transcripts, and control information between the client and server during Ultravox calls. These messages work across WebRTC data channels and WebSocket connections.
+
+## Message Structure
+
+All messages are JSON objects containing:
+- A required `type` field identifying the message type
+- Additional fields specific to each message type
+- Messages follow camelCase naming convention
+
+## Client Messages
+
+Messages sent from client to server:
+
+### Input Messages
+- **InputTextMessage**
+  - `type: "input_text_message"`
+  - `text`: Text content from user
+
+### Control Messages
+- **Ping**
+  - `type: "ping"` 
+  - `timestamp`: Client timestamp for latency measurement
+
+- **SetOutputMedium**
+  - `type: "set_output_medium"`
+  - `medium`: Either "voice" or "text"
+
+### Tool Messages
+- **ClientToolResult**
+  - `type: "client_tool_result"`
+  - `invocation_id`: Matches corresponding invocation
+  - `result`: Tool execution result (optional)
+  - `response_type`: Defaults to "tool-response"
+  - `error_type`: "undefined" or "implementation-error" (optional)
+  - `error_message`: Error details if failed (optional)
+
+## Server Messages
+
+Messages sent from server to client:
+
+### Transcript Messages
+- **Transcript**
```

Comment:
Related to my comment about some overview information for each message, it will eventually be useful to mention the API version in which a message was introduced.  (Probably not worth documenting version 0, so we don't need this yet but worth keeping in mind.)

Reply:
ok sounds good